cloud-aws/src/test/java/com/sequenceiq/cloudbreak/cloud/aws/service/subnetselector/SubnetSelectorServiceTest.java

```diff
@@ -0,0 +1,329 @@
+package com.sequenceiq.cloudbreak.cloud.aws.service.subnetselector;
+
+
+import static com.sequenceiq.cloudbreak.cloud.aws.service.subnetselector.SubnetBuilder.AZ_A;
+import static com.sequenceiq.cloudbreak.cloud.aws.service.subnetselector.SubnetBuilder.AZ_B;
+import static com.sequenceiq.cloudbreak.cloud.aws.service.subnetselector.SubnetBuilder.AZ_C;
+import static com.sequenceiq.cloudbreak.cloud.aws.service.subnetselector.SubnetBuilder.AZ_D;
+import static org.hamcrest.Matchers.hasSize;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertThat;
+import static org.junit.Assert.assertTrue;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.junit.MockitoJUnitRunner;
+
+import com.sequenceiq.cloudbreak.cloud.model.CloudSubnet;
+
+@RunWith(MockitoJUnitRunner.class)
+public class SubnetSelectorServiceTest {
+
+    private final SubnetSelectorService subnetSelectorService = new SubnetSelectorService();
+
+    @Test
+    public void testCollectOnePrivateSubnetPerAzWhenEmpty() {
+        List<CloudSubnet> cloudSubnets = new SubnetBuilder()
+                .build();
+
+        Map<String, CloudSubnet> privateSubnetsPerAz = subnetSelectorService.collectOnePrivateSubnetPerAz(cloudSubnets, 3);
+
+        assertTrue(privateSubnetsPerAz.isEmpty());
+    }
+
+    @Test
+    public void testCollectOnePrivateSubnetPerAz() {
+        List<CloudSubnet> cloudSubnets = new SubnetBuilder()
+                .withPrivateSubnet(AZ_A)
+                .withPrivateSubnet(AZ_B)
+                .withPrivateSubnet(AZ_C)
+                .build();
+
+        Map<String, CloudSubnet> privateSubnetsPerAz = subnetSelectorService.collectOnePrivateSubnetPerAz(cloudSubnets, 3);
+
+        assertThat(privateSubnetsPerAz.values(), hasSize(3));
+    }
+
+    @Test
+    public void testCollectOnePrivateSubnetPerAzWhenMoreThanThree() {
+        List<CloudSubnet> cloudSubnets = new SubnetBuilder()
+                .withPrivateSubnet(AZ_A)
+                .withPrivateSubnet(AZ_B)
+                .withPrivateSubnet(AZ_C)
+                .withPrivateSubnet(AZ_D)
+                .build();
+
+        Map<String, CloudSubnet> privateSubnetsPerAz = subnetSelectorService.collectOnePrivateSubnetPerAz(cloudSubnets, 3);
+
+        assertThat(privateSubnetsPerAz.values(), hasSize(3));
+    }
+
+    @Test
+    public void testCollectOnePrivateSubnetPerAzWhenTwoInOneAZ() {
```

Comment:
I think these test are great but would be beneficial to say what is happenning like testCollectOnePrivateSubnetPerAzWhenTwoInOneAZ -> testCollectOnePrivateSubnetPerAzWhenTwoInOneAZThenShouldReturnOneAZ

Reply:
I agree, I like the test-when-then naming convention. However, without some separators the test names would become too long to read and so just skipped it for brevity. I would prefer some java "illegal" conventions like
testMethodname_whenCondition_thenExpectation
to make it readable.

Reply:
I mostly using this testMethodnameWhenConditionThenExpectation